### ISSUE TITLE
needrestart always lists firewalld.service and user@1000.service

### ISSUE THREAD

**Issue by BjoernDaase** [liske/needrestart#175]
Fedora 31  
needrestart 3.4

On a freshly rebooted Fedora 31, `sudo needrestart -p -l` always reports  
```
WARN - Services: 2 (!), Containers: none, Sessions: none|Services=2;;0;0 Containers=0;;0;0 Sessions=0;0;;0
Services:
- firewalld.service
- user@1000.service
```

And even after restarting one of them (e.g. `systemctl restart firewalld.service`) firewalld gets listed again when running `sudo needrestart -p -l`.

**Comment by liske** [comment]
Could you please provide the output of `needrestart -v`?

**Comment by BjoernDaase** [comment]
Sure (I just restarted the machine)
```
➜  ~ sudo needrestart -v   
[main] eval /etc/needrestart/needrestart.conf
[main] needrestart v3.4
[main] running in root mode
[Core] Using UI 'NeedRestart::UI::stdio'...
[main] systemd detected
[main] #840 uses deleted /tmp/ffiR9RirH
[main] #840 is not a child
[main] #1618 uses deleted /tmp/ffiY17gPy
[main] #1618 is a child of #1493
[main] #840 exe => /usr/bin/python3.7
[Core] #840 is a NeedRestart::Interp::Python
[Core] #840 source is /usr/sbin/firewalld
[main] trying systemctl status
[main] #840 is firewalld.service
[main] #1493 exe => /usr/lib/systemd/systemd
[main] trying systemctl status
[main] #1493 is user@1000.service
[uCode/Intel] current revision: 0xca
++ iucode_tool --scan-system
++ grep -oE '[^[:space:]]+$'
+ sig=0x000806ea
+ '[' -r /sys/devices/system/cpu/cpu0/microcode/processor_flags ']'
++ cat /sys/devices/system/cpu/cpu0/microcode/processor_flags
+ filter='-s 0x000806ea,0x80'
+ type bsdtar
+ IUCODE_TOOL_EXTRA_OPTIONS=
+ test -r /etc/default/intel-microcode
+ . /etc/default/intel-microcode
++ IUCODE_TOOL_EXTRA_OPTIONS=--ignore-broken
+ test '' = no
+ '[' -r '/usr/share/misc/intel-microcode*' ']'
+ iucode_tool -l -s 0x000806ea,0x80 --ignore-broken --ignore-broken -tb /lib/firmware/intel-ucode
+ grep 0x000806ea
[uCode/Intel] available revision: 0x00ca
[Kernel] Linux: kernel release 5.4.18-200.fc31.x86_64, kernel version #1 SMP Fri Feb 7 14:50:05 UTC 2020
[Kernel/Linux] /boot/vmlinuz-5.4.18-200.fc31.x86_64 => 5.4.18-200.fc31.x86_64 (mockbuild@bkernel03.phx2.fedoraproject.org) #1 SMP Fri Feb 7 14:50:05 UTC 2020 [5.4.18-200.fc31.x86_64]*
[Kernel/Linux] /boot/vmlinuz-5.4.17-200.fc31.x86_64 => 5.4.17-200.fc31.x86_64 (mockbuild@bkernel04.phx2.fedoraproject.org) #1 SMP Sat Feb 1 19:00:13 UTC 2020 [5.4.17-200.fc31.x86_64]
[Kernel/Linux] /boot/vmlinuz-5.4.15-200.fc31.x86_64 => 5.4.15-200.fc31.x86_64 (mockbuild@bkernel04.phx2.fedoraproject.org) #1 SMP Tue Jan 28 09:08:32 UTC 2020 [5.4.15-200.fc31.x86_64]
[Kernel/Linux] /boot/vmlinuz-0-rescue-995443f1f6d243e787e3f4243cd8d44b => 5.3.7-301.fc31.x86_64 (mockbuild@bkernel03.phx2.fedoraproject.org) #1 SMP Mon Oct 21 19:18:58 UTC 2019 [5.3.7-301.fc31.x86_64]
[Kernel/Linux] using RPM version sorting
[Kernel/Linux] Expected linux version: 5.4.18-200.fc31.x86_64

Running kernel seems to be up-to-date.

The processor microcode seems to be up-to-date.

Restarting services...
Services to be restarted:
Restart «firewalld.service»? [Ynas?] n
Restart «user@1000.service»? [yNas?] n
Service restarts being deferred:
 systemctl restart firewalld.service
 systemctl restart user@1000.service

No containers need to be restarted.

No user sessions are running outdated binaries.
```

**Comment by liske** [comment]
This happens due to some executable mapped files from `/tmp` which are removed. This should be already fixed for the upcoming release 3.5 by d44354cacde685e0b2a502c6bdf5def23821cc0b.

**Comment by BjoernDaase** [comment]
Alright and thanks. Sorry for the noise :)
